packages/collector/test/tracing/database/couchbase/app.js

```diff
@@ -0,0 +1,658 @@
+    res.json({ result: result.exists });
+  });
+});
+
+app.post('/getAndTouch-promise', async (req, res) => {
+  await collection1.getAndTouch('getandtouch-key-1', 5);
+  res.json({ success: true });
+});
+
+app.post('/getAndTouch-callback', (req, res) => {
+  collection1.getAndTouch('getandtouch-key-2', 5, err => {
+    if (err) return res.status(500).send({ err: err.message });
+    res.json({ success: true });
+  });
+});
+
+app.post('/replace-promise', async (req, res) => {
+  await collection1.replace('replace-key-1', 'replacedvalue');
+  const result = await collection1.get('replace-key-1');
+  res.json({ result: result.value });
+});
+
+app.post('/replace-callback', (req, res) => {
+  collection1.replace('replace-key-1', 'replacedvalue', err => {
+    if (err) return res.status(500);
+
+    collection1.get('replace-key-1', (err2, result) => {
+      if (err2) return res.status(500).send({ err: err.message });
+      res.json({ result: result.value });
+    });
+  });
+});
+
+app.get('/lookupIn-promise', async (req, res) => {
+  const result = await collection1.lookupIn('lookup-key-1', [couchbase.LookupInSpec.get('bar')]);
+  res.json({ result: result.content[0].value });
+});
+
+app.get('/lookupIn-callback', (req, res) => {
+  collection1.lookupIn('lookup-key-1', [couchbase.LookupInSpec.get('bar')], (err, result) => {
+    if (err) return res.status(500).send({ err: err.message });
+    res.json({ result: result.content[0].value });
+  });
+});
+
+app.post('/mutateIn-promise', async (req, res) => {
+  await collection1.mutateIn('mutate-key-1', [couchbase.MutateInSpec.increment('foo', 3)]);
+  res.json({ success: true });
+});
+
+app.post('/mutateIn-callback', (req, res) => {
+  collection1.mutateIn('mutate-key-1', [couchbase.MutateInSpec.increment('foo', 3)], err => {
+    if (err) return res.status(500).send({ err: err.message });
+
+    res.json({ success: true });
+  });
+});
+
+app.post('/insert-promise', async (req, res) => {
+  await collection1.insert('insert-key-5', { name: 'insertvalue' });
+
+  res.json({ success: true });
+});
+
+app.post('/insert-callback', (req, res) => {
+  collection1.insert('insert-key-6', { name: 'insertvalue' }, err => {
+    if (err) return res.status(500).send({ err: err.message });
+
+    res.json({ success: true });
+  });
+});
+
+app.post('/upsert-promise', async (req, res) => {
+  await collection1.upsert('upsert-key-1', { name: 'upsertvalue' });
+
+  res.json({ success: true });
+});
+
+app.post('/upsert-callback', (req, res) => {
+  collection1.upsert('upsert-key-1', { name: 'upsertvalue' }, err => {
+    if (err) return res.status(500).send({ err: err.message });
+
+    res.json({ success: true });
+  });
+});
+
+app.post('/remove-promise', async (req, res) => {
+  const err = req.query.error;
+
+  if (err) {
+    try {
+      await collection1.remove();
+      res.status(500);
+    } catch (removeErr) {
+      res.json({ success: true });
```

Comment:
_suggestion:_ Instead of returning a generic success: true, we could actually return the error message in the HTTP response. Then the test could verify that the expected error message has been returned, and our instrumentation did not accidentally swallow the error. This refers to the one blocking comment from me below, where I suspect that we might swallow errors accidentally.

Reply:
We test if the error message is in the span. I found that useful enough

> error: apiType === 'promise' ? 'invalid argument' : 'document not found'

But I will add it :)